src/Sentry.Unity/UnityNativeScopeObserver.cs

```diff
@@ -0,0 +1,76 @@
+using System.Runtime.InteropServices;
+using Sentry.Extensibility;
+
+namespace Sentry.Unity
+{
+    public class UnityNativeScopeObserver : IScopeObserver
```

Comment:
This will need to stay out of `Sentry.Unity` because the linker expects a native method declared in this file to exist and it won't in some cases (i.e Android, webGL etc)
We'll either need `Sentry.Unity.iOS` or this file goes as a single script